From 8a27b7c118d9092353823e38c3bdaf2b5d3f35a0 Mon Sep 17 00:00:00 2001
From: Ron Shimshock <ron@shimshockgroup.com>
Date: Mon, 22 Feb 2016 06:01:25 -0600
Subject: [PATCH] Fix days to match Issue #1137062 change

---
 includes/calendar_plugin_style.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/calendar_plugin_style.inc b/includes/calendar_plugin_style.inc
index bb19af4..26d2d8d 100644
--- a/includes/calendar_plugin_style.inc
+++ b/includes/calendar_plugin_style.inc
@@ -724,7 +724,7 @@ class calendar_plugin_style extends views_plugin_style {
                   // If the last day of this event exceeds the end of the current month or week,
                   // truncate the remaining days
                   $diff =  $this->curday->difference($this->date_info->max_date, 'days');
-                  $remaining_days = ($this->date_info->granularity == 'month') ? min(6 - $wday, $diff) : $diff - 1;
+                  $remaining_days = ($this->date_info->granularity == 'month') ? min(6 - $wday, $diff) : $diff;
                   // The bucket_cnt defines the colspan.  colspan = bucket_cnt + 1
                   $days = $this->curday->difference($item->date_end, 'days');
                   $bucket_cnt = max(0, min($days, $remaining_days));
-- 
2.7.1

