diff --git a/advpoll.module b/advpoll.module
index b43fee1..462fe94 100644
--- a/advpoll.module
+++ b/advpoll.module
@@ -495,25 +495,48 @@ function advpoll_display_results($nid, $data, $page = 0) {
         $show_bar = _advpoll_show_writeins_access();
       }
 
-      if ($show_bar) {
-        $bars .= theme('advpoll_bar', array(
-          'title' => $title,
-          'percentage' => $item['percentage'],
-          'votes' => $item['votes'],
-          'voted' => $voted,
-          )
-        );
-      }
-    }
-
-    $output .= theme('advpoll_results', array(
-      'bars' => $bars,
-      'total' => $results['total'],
-      'voted' => $votes,
-      'nid' => $nid,
-      'cancel_form' => $rendered_form,
-      )
-    );
+       // Get the total votes for all write in
+       $count = 1;
+       $where_statement = '';
+       $writein_percentage = '';
+       $writein_total = '';
+       foreach ($data->choices as $choice) {
+         if ($choice['write_in'] == 1) {
+           if ($count == 1) {
+             $where_statement .= 'tag=\''. $choice['choice_id'] . '\'';
+           } else {
+             $where_statement .= ' OR tag=\'' . $choice['choice_id'] . '\'';
+           }
+           $count++;
+         }
+       }
+       $result = db_query('SELECT SUM(value) as count FROM votingapi_cache WHERE (' . $where_statement . ') AND function=\'count\'');
+       foreach ($result as $record) {
+         $writein_percentage = ceil((($record->count)*100)/$results['total']);
+         $writein_total = $record->count;
+       }
+ 
+       if ($show_bar) {
+         $bars .= theme('advpoll_bar', array(
+           'title' => $title,
+           'percentage' => $item['percentage'],
+           'votes' => $item['votes'],
+           'voted' => $voted,
+           )
+         );
+       }
+     }
+ 
+     $output .= theme('advpoll_results', array(
+       'bars' => $bars,
+       'total' => $results['total'],
+       'voted' => $votes,
+       'nid' => $nid,
+       'cancel_form' => $rendered_form,
+       'writein_percentage' => $writein_percentage,
+       'writein_total' => $writein_total,
+       )
+     );
   }
 
   return $output;
